cartographer/mapping/2d/grid_2d.h

```diff
@@ -21,12 +21,17 @@
 
 #include "cartographer/mapping/2d/map_limits.h"
 #include "cartographer/mapping/2d/proto/grid_2d.pb.h"
+#include "cartographer/mapping/2d/proto/submaps_options_2d.pb.h"
+#include "cartographer/mapping/grid.h"
 #include "cartographer/mapping/proto/submap_visualization.pb.h"
 
 namespace cartographer {
 namespace mapping {
 
-class Grid2D {
+proto::GridOptions2D CreateGridOptions2D(
+    common::LuaParameterDictionary* const parameter_dictionary);
+
+class Grid2D : public Grid {
```

Comment:
consistency: Does Grid3D also inherit from Grid? (maybe in a later PR)

Reply:
On the long term, yes. Grid is intended as an Interface for Grid2D and a later introduced Grid3D to enable a unification of Submap2D/3D.